new-dti-website/components/team/MemberDisplay.tsx

```diff
@@ -6,28 +6,34 @@ import teamRoles from './data/roles.json';
 import roleIcons from './data/roleIcons.json';
 import { populateMembers } from '../../src/utils/memberUtils';
 
-import alumniMembers from '../../../backend/src/members-archive/fa21.json';
+import alumniMembers from './data/alumni.json';
 
 const allMembers = members as IdolMember[];
 
-const roles = populateMembers(
-  teamRoles as {
-    [key: string]: {
-      roleName: string;
-      description: string;
-      members: IdolMember[];
-      roles: string[];
-      color: string;
-    };
-  },
-  allMembers
-);
+type RoleEntry = {
+  [key: string]: {
+    roleName: string;
+    description: string;
+    members: IdolMember[];
+    roles: string[];
+    color: string;
+  };
+};
+
+const roles = populateMembers(teamRoles as RoleEntry, allMembers);
 
 const MemberDisplay: React.FC = () => {
   const [selectedRole, setSelectedRole] = useState<string>('Full Team');
   const [selectedMember, setSelectedMember] = useState<IdolMember | undefined>(undefined);
   const [clickedSection, setClickedSection] = useState<string | undefined>(undefined);
 
+  // Make roles in alumni section the same order as full team
+  const alumniRoles = populateMembers(teamRoles as RoleEntry, alumniMembers as IdolMember[]);
```

Comment:
What are your thoughts on maybe using a **Set** to track members for the duplicate checks instead of using '.includes'? 
 
`const orderedAlumni: IdolMember[] = [];`
`const seenMembers = new Set<IdolMember>();`

Reply:
I thought about that. My opinion is that the imperative approach plus having to maintain another data structure might be harder to read and also overkill for a section that will only have ~20-40 elements. I usually save such optimizations for larger data sets/backend logic.